temporal-object/README.md

```diff
@@ -0,0 +1,167 @@
+--- # this is so-called 'YAML front matter', read up on it here: http://jekyllrb.com/docs/frontmatter/
+layout: pattern
+title: Temporal Object
+folder: temporal-object 
+permalink: /patterns/temporal-object /
+categories:
+- creational
+  language: en
+  tags:
+- Data Access 
+---
+
+## Name / classification
+
+Temporal Object.
+
+## Also known as
+
+Version History.
+
+## Intent
+
+Describing an object that changes over time.
+
+## Explanation
+
+Real-world example
+
+> A person has a credit card, but the contract associated can change over time, and in fact, the 
+> contract determines most of the details about the card. So, the credit card can be considered 
+> a temporal object, with the contract being Temporal.
+
+In plain word
+
+> An object is largely determined by Temporal attributes, with a history associated to the 
+> versions as they change. 
+
+Martin Fowler says
+
+>This pattern is very common, yet appears in a number of forms. To help grapple with these forms,
+> I find it very helpful to use some role analysis. The pattern essentially has two roles: one 
+> continuity and several versions.
+>
+> Each version captures the state of the object for some period of time. Any time the value of any 
+> property of the object changes, you get a new version. Thus you can imagine the versions as a 
+> list of objects with an Effectivity to handle the date range.  
+>
+>The continuity represents the on going object that goes through these changes in version. It's 
+> the object people refer to when they are thinking of the object throughout its changes.
+
+**Programatic Example**
+
+In this example implementation, `CreditCard`s have contracts that change over time, and most of 
+the card's properties are determined by the contract. 
+
+Here are the relevant parts of the card, note that there are no variables set, and only a 
+history in this example.
+
+```java
+public class CreditCard {
+  private final TemporalCollection<CreditContractVersion> contractHistory;
+
+  public CreditCard(CreditContractVersion contract, SimpleDate effectiveDate) {
+    contractHistory = new TemporalCollection<>();
+    contractHistory.put(effectiveDate, contract);
+  }
+  
+  public CreditContractVersion getContract(SimpleDate date) {
+    return contractHistory.get(date);
+  }
+  
+  public void addContract(CreditContractVersion contract, SimpleDate effectiveDate) {
+    contractHistory.put(effectiveDate, contract);
+  }
+
+  public boolean isExpired() {
+    // if expiration date before current date, return true
+    return getContract().getExpirationDate().compareTo(SimpleDate.getToday()) < 0;
+  }
+
+  public int getNumber() {
+    return getContract().getCardNumber();
+  }
+  //...
```

Comment:
What is the purpose of the `//...` in this example?

Reply:
I think this is in error. Thank you for pointing it out.